python/graphstorm/model/gat_encoder.py

```diff
@@ -141,6 +141,120 @@ def forward(self, g, inputs):
+            including both node and edge features.
+        
+        from graphstorm.model import GATConv
+
+        layer = GATConvwithEdgeFeat(h_dim, h_dim, num_heads, num_ffn_layers_in_gnn)
+        h = layer(g, input_feature)
+
+    Parameters
+    ----------
+    in_feat: int
+        Input feature size.
+    out_feat: int
+        Output feature size.
+    num_heads: int
+        Number of heads in Multi-head attention.
+    activation: callable, optional
+        Activation function. Default: relu
+    dropout: float, optional
+        Dropout rate. Default: 0.0
+    bias: bool, optional
+        True if bias is added. Default: True
+    num_ffn_layers_in_gnn: int, optional
+        Number of layers of ngnn between gnn layers. Default: 0
+    ffn_actication: torch.nn.functional, optional
+        Activation Method for ngnn. Default: relu
+    """
+    def __init__(self,
+                 in_feat,
+                 out_feat,
+                 num_heads,
+                 activation=F.relu,
+                 dropout=0.0,
+                 bias=True,
+                 num_ffn_layers_in_gnn=0,
+                 ffn_activation=F.relu):
+        super(GATConvwithEdgeFeat, self).__init__()
+        # with the edge and node encoder, node_prefix_shape and edge_prefix_shape are determined
+        edge_feats = in_feat
+        self.conv = dglnn.EdgeGATConv(
+            in_feat, edge_feats, out_feat // num_heads, num_heads, dropout,
+            activation=activation, allow_zero_in_degree=True,
+            bias=bias
+        )
+
+        # ngnn
+        self.num_ffn_layers_in_gnn = num_ffn_layers_in_gnn
+        self.ngnn_mlp = NGNNMLP(out_feat, out_feat,
+                                 num_ffn_layers_in_gnn, ffn_activation, dropout)
+
+    def forward(self, g, inputs):
+        """ GAT layer forward computation.
+
+        Parameters
+        ----------
+        g: DGLHeteroGraph
+            Input DGL heterogenous graph.
+        inputs: dict of Tensor
+            Node features for the default node type in the format of
+            {``dgl.DEFAULT_NTYPE``: tensor, ``dgl.DEFAULT_ETYPE``: tensor}. 
+            The definition of ``dgl.DEFAULT_NTYPE`` and ``dgl.DEFAULT_ETYPE`` can
+            be found at `DGL official Github site <https://github.com/dmlc/dgl/blob/
+            cb4604aca2e9a79eb61827a71f1f781b70ceac83/python/dgl/distributed/constants.py#L8>`_.
+
+        Returns
+        -------
+        dict of Tensor: New node embeddings for the default node type in the format of
+        {``dgl.DEFAULT_NTYPE``: tensor}. The definition of ``dgl.DEFAULT_NTYPE`` can
+        be found at `DGL official Github site 
+        <https://github.com/dmlc/dgl/blob/cb4604aca2e9a79eb61827a71f1f781b70ceac83/
+        python/dgl/distributed/constants.py#L8>`_.
+        """
+
+        # add self-loop during computation.
+        assert DEFAULT_NTYPE in inputs and DEFAULT_ETYPE in inputs, \
+            "GAT encoder only support homogeneous graph."
```

Comment:
Can we also add the error message that both node feature and edge feature need to exist.